2020/day3.clj

```diff
@@ -7,27 +7,34 @@
        (string/split-lines)))
 
 (defn next-location
-  [loc [right down]]
-  [(+ (first loc) right)
-   (+ (second loc) down)])
+  [[row col] [right down]]
+  [(+ row right)
+   (+ col down)])
 
 (defn locations-s
-  [height [right down]]
-  (take-while #(< (second %) height)
-              (map #(next-location [0 0] [(* right %) (* down %)])
-                   (range))))
+  [slope]
+  (map (fn [n] (next-location [0 0] [(* (first slope) n) (* (second slope) n)]))
```

Comment:
There's no need to compress this to 1 line

Reply:
rewrote it.